include/neml2/models/crystallography/SymmetryOperator.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include "neml2/tensors/R2Base.h"
+
+namespace neml2
+{
+class Rot;
+class Quaternion;
+
+namespace crystallography
+{
+/// @brief A generic symmetry operator
+/// This includes the identity, proper rotations, improper rotations,
+/// and inversion centers.  We also include reflections, even though
+/// they are not proper operators themselves as they are useful in
+/// defining others.
+class SymmetryOperator : public R2Base<SymmetryOperator>
```

Comment:
Can you remind me why we need `SymmetryOperator` to be a class, instead of just `R2`?

Reply:
Because in theory we could implement special versions for SR2, WR2, etc (or even vec for some of the operations like reflection and rotation).  But I didn't end up doing that for `Vec` and I don't know if we ever will need to do it...  So I can refactor to just make these R2s.  We will still need a `Transformable` mixin thing because transforming an R2 is a different operation than a R4, etc...

Reply:
I got rid of it.